packages/xdl/src/gating/FeatureGating.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  featureGateDefaultValueWhenNoServerValue,
+  FeatureGateEnvOverrides,
+  FeatureGateKey,
+  FeatureGateTestOverrides,
+} from '../internal';
+
+export default class FeatureGating {
+  constructor(
+    private readonly serverValues: { [key: string]: boolean },
+    private readonly envOverrides: FeatureGateEnvOverrides
+  ) {}
+
+  public isEnabled(experimentKey: FeatureGateKey): boolean {
+    if (process.env.NODE_ENV === 'test' && FeatureGateTestOverrides.isOverridden(experimentKey)) {
```

Comment:
This seems like an odd thing to ship to production. Couldn't the same be achieved using a jest mock on this function?

Reply:
In the CLIs yep, I think it could. I think I kept it this way so that all of our feature gating infra was the same across all our codebases (this is the way the test mocking works in www, website, eas-cli).